catalog/format/iceberg/src/main/java/org/projectnessie/catalog/formats/iceberg/nessie/NessieModelIceberg.java

```diff
@@ -709,19 +730,56 @@ public static NessiePartitionStatisticsFile icebergPartitionStatisticsFileToNess
         icebergPartitionStatisticsFile.fileSizeInBytes());
   }
 
-  private static Map<Integer, NessieField> collectAllSchemaFields(List<NessieSchema> schemas) {
+  /**
+   * Collects all fields, top-level and nested in structs, from the given schema into the given map.
+   *
+   * <p>Similar to {@link #collectFieldsByIcebergId(NessieStruct, Map)}.
+   */
+  public static void collectFieldsByNessieId(
+      NessieSchema schema, Map<UUID, NessieField> targetFieldMap) {
+    collectFieldsByNessieId(schema.struct(), targetFieldMap);
+  }
+
+  /**
+   * Collects all fields, top-level and nested in structs, from the given struct into the given map.
+   *
+   * <p>Similar to {@link #collectFieldsByIcebergId(NessieStruct, Map)}.
+   */
+  public static void collectFieldsByNessieId(
+      NessieStruct struct, Map<UUID, NessieField> targetFieldMap) {
+    for (var field : struct.fields()) {
+      targetFieldMap.put(field.id(), field);
+      var type = field.type();
+      if (requireNonNull(type.type()) == NessieType.STRUCT) {
```

Comment:
I'm not sure, but I think you also need to inspect MAP and LIST – see my comment in tests.

Reply:
Nope. Fields in maps and lists can't be referred from partition-specs and sort-orders.

Reply:
Added a comment